lib/serve_html_task.rb

```diff
@@ -16,13 +16,17 @@
 # you may find current contact information at www.suse.com
 
 class ServeHtmlTask
-  def serve(system_description_store, opts)
-    hostname = Socket.gethostbyname(Socket.gethostname).first
-    url = "http://#{hostname}:#{opts[:port]}/"
-    if opts[:public]
-      remote_ip = Socket.ip_address_list.find { |ip| ip.ipv4? && !ip.ipv4_loopback? }.ip_address
-      remote_url = "http://#{remote_ip}:#{opts[:port]}/"
+  def assemble_url(opts)
+    host = if opts[:public]
+      Socket.gethostname
```

Comment:
I guess your suggestion is a little closer to what I did before, but it was decided against doing so in the end. If it's still unclear maybe we have to discuss separately how we want to proceed.